src/main/java/com/blamejared/compat/forestry/CharcoalPile.java

```diff
@@ -0,0 +1,155 @@
+	}
+
+	private static class Add extends BaseCharcoal {
+
+		private int amount;
+
+		protected Add(Block block, int amount) {
+			super(block);
+			this.amount = amount;
+		}
+
+		protected Add(IBlockState state, int amount) {
+			super(state);
+			this.amount = amount;
+		}
+
+		@Override
+		public void apply() {
+			if (manager == null) {
+				CraftTweakerAPI.logError("Charcoal manager is null, probably the charcoal module is not enabled");
+				return;
+			}
+			if (block != null) {
+				manager.registerWall(block, amount);
+			} else if (state != null) {
+				manager.registerWall(state, amount);
+			} else {
+				CraftTweakerAPI.logError("Both block and blockstate were null, no wall registered");
+			}
+		}
+	}
+
+	@ZenMethod
+	public static void removePile(IBlock block) {
+		ModTweaker.LATE_REMOVALS.add(new Remove(asBlock(block)));
+	}
+
+	@ZenMethod
+	public static void removePileState(crafttweaker.api.block.IBlockState state) {
+		ModTweaker.LATE_REMOVALS.add(new Remove(asBlockState(state)));
+	}
+
+	@ZenMethod
+	public static void removePileStack(IItemStack stack) {
+		ModTweaker.LATE_REMOVALS.add(new Remove(asBlock(stack.asBlock())));
+	}
+
+	private static class Remove extends BaseCharcoal {
+
+		protected Remove(Block block) {
+			super(block);
+		}
+
+		protected Remove(IBlockState state) {
+			super(state);
+		}
+
+		@Override
+		public void apply() {
+			if (manager == null) {
+				CraftTweakerAPI.logError("Charcoal manager is null, probably the charcoal module is not enabled");
+				return;
+			}
+			boolean success = false;
+			if (block != null) {
+				success = manager.removeWall(block);
+			} else if (state != null) {
+				success = manager.removeWall(state);
+			} else {
+				CraftTweakerAPI.logError("Both block and blockstate were null, no wall removed");
+			}
+			if (!success) {
+				CraftTweakerAPI.logError("Removing wall was not successful");
+			}
+		}
+	}
+
+	private static class BaseCharcoal extends BaseAction {
+
+		protected static ICharcoalManager manager = TreeManager.charcoalManager;
+		protected Block block;
+		protected IBlockState state;
+
+		protected BaseCharcoal(Block block) {
+			super(CharcoalPile.NAME);
+			this.block = block;
+		}
+
+		protected BaseCharcoal(IBlockState state) {
+			super(CharcoalPile.NAME);
+			this.state = state;
+		}
+
+		@Override
+		public void apply() {
+
+		}
+	}
+
+
+	// helpers, here for now
+	private static Block asBlock(IBlock block) {
+		Object internal = block.getDefinition().getInternal();
+		if (!(internal instanceof Block)) {
+			CraftTweakerAPI.logError("Not a valid block: " + block);
+		}
+		return (Block) internal;
+	}
+
+	private static IBlockState asBlockState(crafttweaker.api.block.IBlockState blockstate) {
+		Object internal = blockstate.getInternal();
+		if (!(internal instanceof IBlockState)) {
+			CraftTweakerAPI.logError("Not a valid blockstate: " + blockstate);
+		}
+		return (IBlockState) internal;
```

Comment:
You know that this will throw a ClassCastException if the condition for `logError` is met?
Why not return `null` in that instanceof Check?